pulp_rpm/app/access_policy.py

```diff
@@ -0,0 +1,39 @@
+from logging import getLogger
+
+from pulpcore.plugin.models import RepositoryVersion
+from pulpcore.plugin.access_policy import AccessPolicyFromDB
+
+from pulp_rpm.app.models.repository import RpmRepository
+
+_logger = getLogger(__name__)
+
+
+class CopyAccessPolicy(AccessPolicyFromDB):
+    """
+    Access policy for ContainerNamespaceViewSet.
+    """
+
+    def has_perms_to_copy(self, request, view, action):
+        """
+        Check if the source and destination repository matches the usernames permissions.
+        """
+        serializer = view.serializer_class(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+
+        dest_repo_id = request.data["config"][0]["dest_repo"].split("/")[-2]
```

Comment:
i believe you would also find dest_repo in the `view.kwargs`

Reply:
Unfortunately I won't, but imho still better idea to parse serializer data that the request itself.